### PR TITLE
Add tests regarding pod ownership

### PR DESCRIPTION
**Pull request by adleong**
https://github.com/linkerd/linkerd2/pull/5567 introduced behavior where we consider the owner references of a pod to be authoritative about which resources contain that pod.  This is in contrast to relying on the label selectors which can be ambiguous when, for example, deployments have overlapping label selectors.

This distinction is non-obvious.  To prevent regressions in this behavior, we add tests to codify this requirement.

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

File: pkg/k8s/portforward.go [hunk 1]
```diff
@@ -124,7 +124,14 @@ func newPortForward(
 	emitLogs bool,
 ) (*PortForward, error) {
 
-	req := k8sAPI.CoreV1().RESTClient().Post().
+	restClient := k8sAPI.CoreV1().RESTClient()
+	if fakeRest, ok := restClient.(*rest.RESTClient); ok {
+		if fakeRest == nil {
+			return nil, nil
+		}
+	}
```
Comment (siggy) [inline pkg/k8s/portforward.go:132]: if ok == false, i assume something else is broken, perhaps we return an error in that case?

File: pkg/k8s/portforward_test.go [hunk 2]
```diff
@@ -69,57 +69,177 @@ spec:
 }
 
 func TestNewPortForward(t *testing.T) {
-	// TODO: test successful cases by mocking out `clientset.CoreV1().RESTClient()`
```
Comment (siggy) [inline pkg/k8s/portforward_test.go:72]: 🙏

**Review by siggy — APPROVED** [review]
nice! 🚢 👍

File: pkg/k8s/portforward.go (hunk 1 above)
Comment (adleong) [reply pkg/k8s/portforward.go]: If ok == false, it just means that the internal implementation of the interface has changed which shouldn't be a problem when this is called in normal operation (not in a test).  I tried to make this as conservative as possible: i.e. only return early if we can be absolutely sure we're in a test.  Added a comment to give more context to this.

**Review by Pothulapati — APPROVED** [review]
LGTM :shipit:

**Review by kleimkuhler — APPROVED** [review]
Awesome thanks for adding this 👍

**Pull request merged**